otelcol/configprovider.go

```diff
@@ -84,7 +84,7 @@ func (cm *configProvider) Get(ctx context.Context, factories Factories) (*Config
 	}
 
 	var cfg *configSettings
-	if cfg, err = unmarshal(conf, factories); err != nil {
+	if cfg, err = Unmarshal(conf, factories); err != nil {
```

Comment:
Do we have to expose this method? It seems like downstream we could use `NewConfigProvider.Get` instead to do the same thing.

Reply:
`ConfigProvider` can only get config from URIs. In our case the Agent parses its own config file, and we can't give the whole file to `ConfigProvider`because it contains non-Otel config. We could theoretically add new functionality to  `ConfigProvider`but I don't think it'll be accepted upstream since what we do here is not really what it is meant to do.

Reply:
I'm not sure I have the full context, but you could have another config provider specific to agent. We have a few config providers already, like the file provider, HTTP provider, and so on. Look at `confmap/provider` for example of implementations.